### PR TITLE
Task contact list part2

### PR DESCRIPTION
**Pull request by MaxGmn**

### CODE REVIEW

File: ContactList/ContactList/UpdateViewController.swift [hunk 1]
```diff
@@ -83,34 +84,19 @@ class UpdateViewController: UIViewController {
...
-                self.present(self.picker, animated: true, completion: nil)
-            }
-            let cameraAction = UIAlertAction(title: "Camera", style: .default){action in
-                self.picker.allowsEditing = false
-                self.picker.sourceType = .camera
-                self.picker.cameraCaptureMode = .photo
-                self.picker.modalPresentationStyle = .fullScreen
-                self.present(self.picker, animated: true, completion: nil)
-        }
+        if imageArea.image != Constants.emptyAvatar {
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/UpdateViewController.swift]: Why don't you check value of `imageState`?

File: ContactList/ContactList/UpdateViewController.swift [hunk 2]
```diff
@@ -157,20 +143,61 @@ class UpdateViewController: UIViewController {
...
+        switch imageState {
+        case .removed:
+            imageArea.image = Constants.emptyAvatar
+        case .changed(let newImage):
+            imageArea.image = newImage
+        default:
+            break
+        }
+        
+        currentPersonCopy.image = imageArea.image!
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/UpdateViewController.swift]: Why don't you use image from imageState? Default image should never be set into models - they are being used for presentation purposes only

File: ContactList/ContactList/UpdateViewController.swift [hunk 3]
```diff
@@ -157,20 +143,61 @@ class UpdateViewController: UIViewController {
         saveButton.isEnabled = (currentPersonForEditing != currentPersonCopy || imageState != ImageEditState.noChanges) && fieldsCheckingIsOk
     }
     
+    func changeCurrentImage(imageState: ImageEditState) {
+        
+        imageArea.contentMode = .scaleAspectFit
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/UpdateViewController.swift]: Must be set once in Interface Builder

File: ContactList/ContactList/UserDefaults.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+//  ContactList
+//
+//  Created by Maksym Humeniuk on 5/14/19.
+//  Copyright © 2019 Maksym Humeniuk. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class UserDefaultsWorking {
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/UserDefaults.swift]: What `Working` means?

File: ContactList/ContactList/UserDefaults.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+        do {
+            return try NSKeyedUnarchiver.unarchiveTopLevelObjectWithData(data as! Data) as! [Person]
+        } catch {
+            return []
+        }
+       
+    }
+    
+    static func saveImage (by imageState: ImageEditState, name: String) {
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/UserDefaults.swift]: Your image saving code doesn't deal with UserDefaults, so it should not be here

File: ContactList/ContactList/UserDefaults.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        case .noChanges:
+            return
+        case .removed:
+            removeFile(fileName: name)
+        case .changed(let image):
+            createFile(fileName: name, image: image)
+        }
+    }
+    
+    static func getImage(fileName: String) -> UIImage {
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/UserDefaults.swift]: This method should return `UIImage?`.
Default image must be set in views only.

File: ContactList/ContactList/ViewControllerForShow.swift [hunk 7]
```diff
@@ -52,7 +51,7 @@ class ViewControllerForShow: UIViewController {
         lastNameLabel.text = person.lastName
         phoneNumberLabel.text = person.phoneNumber
         emailLabel.text = person.email
-        imageArea.image = person.image ?? Constants.emptyAvatar
+        imageArea.image = person.image
```
Comment (oleksii-lubianyi-epam) [inline ContactList/ContactList/ViewControllerForShow.swift]: `imageArea.image = person.image ?? Constants.emptyAvatar`

**Review by oleksii-lubianyi-epam — CHANGES_REQUESTED** [review]

**Review by oleksii-lubianyi-epam — APPROVED** [review]

File: ContactList/ContactList/WorkWithData.swift [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+//  ContactList
+//
+//  Created by Maksym Humeniuk on 5/14/19.
+//  Copyright © 2019 Maksym Humeniuk. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class WorkWithData {
```
Comment (ghost) [inline ContactList/ContactList/WorkWithData.swift:12]: WorkWithData name sounds weird usual something like DataManager is used

**Review by ghost — APPROVED** [review]

**Pull request merged**